content-services/content-sources-backend
# Fixes 4339,4371: add uploads API

## Summary

## Testing steps

1.  Create a repository in the UI or Api
2.   use the ./scripts/upload.py script to upload some repositories
```
IDENTITY_HEADER="eyJpZGVudGl0eSI6eyJvcmdfaWQiOiI5IiwgInR5cGUiOiJVc2VyIiwidXNlciI6eyJ1c2VybmFtZSI6ImZvbyJ9LCJhY2NvdW50X251bWJlciI6ImZvbyIsImludGVybmFsIjp7Im9yZ19pZCI6IjkifX19Cg==" python ./scripts/uploads.py      "8695b221-bf5d-4f5b-b544-43e563e0f558"  ~/rpms/crow-0.8-1.noarch.rpm   ~/rpms/foo-1.2.3.noarch.rpm
```

try with large RPMs, many rpms, etc...
Note, you'll need to make sure you use the same org in the identity header as you've used to create the repo

4. Fetch the repository and check the package_count, fetch the repository rpms  (/repositories/UUID/rpms)  
5. check that a snapshot is made each time, and that the snapshots show the rpms that should be there
6. access the repo in pulp for the snapshots and verify expected rpms are present  

7.  Re-run the command with rpms that have already been uploaded, (or a mix of some that have and haven't), if there are no new rpms, no snapshot should be made

8.  If you want run the command with  -f to finalize the uploads before adding them.  Note that this will fail if the rpms are already present in the org (we likely won't be using this method going forward)

## Checklist
